pkg/boot/kexec/purgatories.go

```diff
@@ -0,0 +1,309 @@
+// Copyright 2022 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+//go:build ignore
+
+package main
+
+type asm struct {
+	name string
+	cc   []string
+	ld   []string
+	code string
+}
+
+var asms = []asm{
+	{
+		name: "to32bit_3000",
```

Comment:
We have a to32bit trampoline to work off of as well -- https://github.com/u-root/u-root/blob/main/pkg/boot/multiboot/internal/trampoline/trampoline_linux_amd64.s

Reply:
Actually, these are the _same_ trampoline. That one is just in Go and allows for values to be read into rbx, rsi, rax before jumping into the new kernel.

I'm happy to get on a call and explain how it works.

Reply:
I'd rather not merge them until we have some certainty around all this. It's still not clear to me that these are sufficient. ONCE we have certainty, then we can look at it. But this is not ready yet.

Reply:
What do you mean by "these are sufficient"? Are you saying these trampolines might not be enough?

My point with my comments here and below was to show that the 32bit trampoline was actually *the same code*. If you read them closely side by side you will see they are. I can tell you why, too: Max took this trampoline originally from the same place you did, and then translated it 1:1 to Go assembly 

I'm trying to save you some work. Don't be defensive about it. I don't care if it happens now or later.

Reply:
That translation to Go assembly: I'm fairly sure it can be done better, which is the other reason I'm not ready to merge. Lessons learned from coreboot: don't merge this kind of code too soon.